src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionXpathSingleFilterTest.java

```diff
@@ -400,6 +401,59 @@ public void testDecideById() throws Exception {
                 expected, removeSuppressed(expected, suppressed));
     }
 
+    /**
+     * This test is required to cover pitest mutation
+     * for reset of 'file' field in SuppressionXpathSingleFilter.
+     * This not possible to reproduce by natural execution of Checkstyle
+     * as config is never changed in runtime.
+     * Projects that use us by api can reproduce this case.
+     * We need to allow users to reset module property to default state.
+     *
+     * @throws Exception when there is problem to load Input file
+     */
+    @Test
+    public void testUpdateFilterSettingsInRunTime() throws Exception {
+        final File file = new File(getPath("InputSuppressionXpathSingleFilterComplexQuery.java"));
+
+        final SuppressionXpathSingleFilter filter = new SuppressionXpathSingleFilter();
+        filter.setChecks("MagicNumber");
+        filter.finishLocalSetup();
+
+        final Violation violation = new Violation(27, 21, TokenTypes.NUM_DOUBLE, "",
+                        "", null, null, null,
+                        MagicNumberCheck.class, null);
+
+        final FileContents fileContents =
+                new FileContents(new FileText(file, StandardCharsets.UTF_8.name()));
+
+        final TreeWalkerAuditEvent ev = new TreeWalkerAuditEvent(fileContents, file.getName(),
+                violation, JavaParser.parseFile(file, JavaParser.Options.WITHOUT_COMMENTS));
+
+        assertWithMessage("match ia expected as 'files' is defaulted")
+                .that(filter.accept(ev))
+                .isFalse();
+
+        // set non-default value for files
+        filter.setFiles(sanitizeWindowsPathForRegexp(file.getPath() + ".never.match"));
+        filter.finishLocalSetup();
+
+        assertWithMessage("no match is expected due to weird value in 'files'")
+                .that(filter.accept(ev))
+                .isTrue();
+
+        // reset files to default value of filter
+        filter.setFiles(null);
+        filter.finishLocalSetup();
```

Comment:
I am pretty sure this goes against the normal lifecycle. We finish the setup 3 separate times. What is the mutation we are trying to kill?

Reply:
Mutation is in diff as removal, in suppression files.

Reply:
I assume this is it.
https://github.com/checkstyle/checkstyle/blob/a89ddac62254f3b94a2d97231bf6a878563d74af/src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionXpathSingleFilter.java#L125

https://pitest.org/quickstart/mutators/#EXPERIMENTAL_MEMBER_VARIABLE
> Please Note: This mutator is likely to create equivalent mutations if a member variable is explicitly initialized with the Java default value for the specific type of the member variable as in

This seems like another example we should report to pitest to change.
````
variable = assignment;
````

````
into
````
if (assignment == null) new RuntimeException();
variable = assignment;
````

Similar to what it does for other mutators like RETURN_VALS.
https://pitest.org/quickstart/mutators/#return-values-mutator-return_vals

Reply:
I interpreted pitest as:
```
public void setFiles(String files) {
        if (files == null) {
            
        }
        else {
            this.files = Pattern.compile(files);
        }
    }
```
and covered ^^^ mutation.

Reply:
Why can't we change this code to:
```
 if (files != null) {
     this.files = Pattern.compile(files);
 }

```
?

Reply:
We try the best to allow user to be able to put mutate move modules to default state of properties.
In other words, user created module, changed values of properties, changed his mind and want to change property value back to default.

it is not possible in XML in most cases where null is default value. But xml config is not whole world for us, I hope one day there will be more formats we support. If our java beans modules continue to allow setting default value, we will always be good.

So this is extreme edge case for us, I doubt much users care about this, but  as a library we should not have opinion on our modules usage, and just allow to set default value back by setter.

Reply:
Do users really consider check module properties mutable? I understand that due to the way we instantiate/configure checks that it is not reasonable to assign all properties via constructor as final fields, but usage of check modules as mutable objects seems like a bad idea to me.

Reply:
Modules were mutable for 20 years.
We do not know how our users use us.
But ideally module should be initialized by explicit default values and work same as without being explicit.